### ISSUE TITLE
Wrong use of Optional[] in typing for config parameter in the Language class

### ISSUE THREAD

**Issue by MichalMalyska** [explosion/spaCy#9012]
By the definition: https://docs.python.org/3/library/typing.html#typing.Optional
it is meant to signify a Union[sth, None] however if you pass a config=None to the add_pipe method:
https://github.com/explosion/spaCy/blob/d94ddd568691cfa9d56b353173237afe1f328b43/spacy/language.py#L725
you will get an error downstream from here:
https://github.com/explosion/spaCy/blob/d94ddd568691cfa9d56b353173237afe1f328b43/spacy/language.py#L628

## Which page or section is this issue related to?
Code documentation in Language Pipeline
https://github.com/explosion/spaCy/blob/d94ddd568691cfa9d56b353173237afe1f328b43/spacy/language.py

**Comment by svlandeg** [comment]
You're right, the code does not allow this parameter to be `Optional`. Its optional in the sense that there's a default value, but the typing is wrong, we'll get that fixed. Thanks for the report!
